src/pytorch_lightning/tuner/lr_finder.py

```diff
@@ -297,6 +298,7 @@ def __lr_finder_restore_params(trainer: "pl.Trainer", params: Dict[str, Any]) ->
     trainer.callbacks = params["callbacks"]
     trainer.logger = params["logger"]
     trainer.fit_loop.max_steps = params["max_steps"]
+    trainer.strategy.setup_optimizers = params["setup_optimizers"]
```

Comment:
Can you add a test to avoid regressions?

Reply:
Thanks for the suggestion. I've extended the existing test `test_trainer_reset_correctly` to also check the attribute `trainer.strategy.setup_optimizers`, which it wasn't doing so far.